source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -554,6 +614,400 @@ namespace aspect
       }
 
 
+      template <int dim>
+      std::pair<std::vector<double>, std::vector<Tensor<2,3>>>
+      CrystalPreferredOrientation<dim>::compute_derivatives_drex_2004(const unsigned int cpo_index,
+                                                                      const ArrayView<double> &data,
+                                                                      const unsigned int mineral_i,
+                                                                      const SymmetricTensor<2,3> &strain_rate_3d,
+                                                                      const Tensor<2,3> &velocity_gradient_tensor,
+                                                                      const std::array<double,4> ref_resolved_shear_stress,
+                                                                      const bool prevent_nondimensionalization) const
+      {
+        SymmetricTensor<2,3> strain_rate_nondimensional = strain_rate_3d;
+        Tensor<2,3> velocity_gradient_tensor_nondimensional = velocity_gradient_tensor;
```

Comment:
Can you move those two linear below the if statement and always do ``= strain_rate_3d / nondimensionalization_value;``?